word_guess_owl.rb

```diff
@@ -0,0 +1,153 @@
+  end
+
+  def winning
+    puts "\n     HOOOOO-RAY!"
+    puts " {^;^}"
+    puts " /)__)"
+    puts "/  ;;"
+  end
+end
+# Allows user to guess letters, validates their input and prompts again if not valid,
+# stores letters already guessed, and increments the number of wrong guesses,
+# generates blank letter spaces, and fills in the blanks as game progresses.
+class Game
+  def initialize (word)
+    @finished_word = word
+    @letters = word.upcase.chars
+    @wrong_guesses = 0
+    @guessed_letters = []
+    @word_array =@letters.length.times.collect {|x| "_"}
+  end
+
+  attr_reader :letters, :finished_word, :guessed_letters, :wrong_guesses, :word_array
+
+  def guess_letter
+    print "Please guess a letter: "
+    guess = gets.chomp.upcase
+    while @guessed_letters.include?(guess)
+      print "You already guessed that letter. Guess again: "
+      guess = gets.chomp.upcase
+    end
+    # only lets the user guess A through Z
+    guess_req = /[A-Z]/
+    # this is making sure that the user guess does not meet our regex requirements or more then one was put in
+    while !guess_req.match(guess) || guess.length > 1
+      print "That is not a correct input. Please guess a letter: "
+      guess = gets.chomp.upcase
+    end
+
+    if @letters.include?(guess)
+      @guessed_letters << guess
+      fill_in(guess)
+      return "That is a correct guess."
+    else
+      @guessed_letters << guess
+      @wrong_guesses += 1
+      return "That is a wrong guess."
+    end
+  end
+
+  def fill_in(guess)
+    @letters.each_with_index do |letter, i|
+      if letter == guess
+        @word_array[i] = guess
+      end
+    end
+    return @word_array
+  end
+
+  def print_word_array
+    printing_word = ""
+    @word_array.each do |letter|
+      printing_word += "#{letter} "
+    end
+    return printing_word
+  end
+
+  def what_was_guessed
+    user_guesses = ""
+    @guessed_letters.each do |letter|
+      user_guesses += letter + " "
+    end
+    return user_guesses
+  end
+end
+
+# generates random word for user to guess
+def word_generator
+  word_options = ["BAZAAR", "BROOD", "EYRIE", "HOOTING", "LOOMING", "NEST", "STOOPING", "BROOD", "DISS", "PAIR", "PARLIAMENT", "SAGACIOUSNESS", "STARE", "WISDOM", "STABLE", "PROHIBITION", "VOLERY", "BLIZZARD"]
+  return word_options.sample
+end
+
+# prompts user at completion of game if they would like to intiate a new game
+def new_game
+  print "\nDo you want to guess another word? Y/N: "
+  answer = gets.chomp.upcase
+  if answer == "Y"
+    replay = true
+  else
+    puts "Thanks for playing with us."
+    exit
+  end
+end
+
+# user interface:
+# continues to prompt user to guess letters until the game is won or lost
+# game is won if guessed letters complete the randomly selected word
+# game is lost if number of guesses exceeds the number of intial zz's
+# prints ASCII art to represent the number of wrong guesses
+# changes ASCII art if user wins or loses
+replay = true
+while replay
```

Comment:
You are right that this could go into the `Game` class as a method since it's integral to the game.